framework/di/Container.php

```diff
@@ -527,7 +531,17 @@ public function resolveCallableDependencies(callable $callback, $params = [])
                 } elseif (isset(Yii::$app) && Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
                     $args[] = $obj;
                 } else {
-                    $args[] = $this->get($className);
+                    // If the argument is optional we catch not instantiable exceptions.
+                    try {
+                        $args[] = $this->get($className);
+                    } catch (NotInstantiableException $e) {
```

Comment:
In this case I am unsure such change is correct.
At present state DI triggers an error if there is no mapping for the interface, thus developer may be aware of it and fix it.
With this change DI just silently skips the injection.